File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> mySet;

        int len = 0;

        int left = 0;

        for (int right = 0; right < s.size(); right++) {
            char c = s[right];

            while(mySet.find(c) != mySet.end()) {
                mySet.erase(s[left++]);
            }

            mySet.insert(c);
            len = max(len, right - left + 1);
            
        }

        return len;


    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.swift
class Solution {
    func lengthOfLongestSubstring(_ s: String) -> Int {
        var lastIndex: [Character:Int] = [:]
        var longest = 0
        var left = 0

        for (right, char) in s.enumerated() {

            if let index = lastIndex[char] {
                left = index + 1
            }

            lastIndex[char] = right


            longest = max(longest, right - left + 1)
        }

        return longest
     }
}